### PR TITLE
Handle package-specific warning suppressions

### LINKED ISSUE #33972: Potential improvements to AdvancedBuild/cibuildwheel
A couple of requests that would make working with the new AdvancedBuild/cibuildwheel a little easier.

- When cibuildwheel fails, the build stage of the pipeline should fail. Currently if cibuildwheel fails, the build stage passes, and the pipeline moves on to test phases. If the build stage is for whatever reason not able to produce the expected package list, it really should fail to indicate to us and the rest of the pipeline that something is not right. Currently I believe the test phases will fail if the build fails, preventing the whole pipeline from passing, but we shouldn't rely on that to know to check the build.
  - If cibuildwheel fails, the build phase will still produce one wheel file (I think it's the wheel for the Python version/platform of the build machine) which can also be a bit misleading.
- It would be nice if the logs from cibuildwheel bubbled up into the pipeline output on ADO. Currently you have to download the log file to see any output from cibuildwheel.
  - Along similar lines, maybe we could bubble up warnings from cibuildwheel as pipeline warnings that show up on the main page for the pipeline run.
- This may be less about AdvancedBuild and more specific to C extension packages but for `azure-storage-extensions` we ignore a ton of tox envs. Each one we ignore generates a warning on the pipeline run. Since this will be the case for ALL Storage pipeline runs going forward, it would be nice if we could suppress those warnings somehow. Perhaps suppress per package or something.

### PR DESCRIPTION
**Pull request by scbedd**
@jalauzon-msft this resolves #33972

- [x] Fixes the issue with crashed build stage not properly failing when `cibuildwheel` fails
- [x] Adds package-specific suppression via glob string array, see new test for example usage
   - This simply suppresses the _warning_ about the environment being skipped. This doesn't affect actual skip logic at all.
- [x] ~~Discovered issue, it's suppressing all warnings for some reason, need to figure out why.~~ All jobs need to complete before warnings are output.

### CODE REVIEW

File: tools/azure-sdk-tools/ci_tools/build.py [hunk 1]
```diff
@@ -180,17 +180,9 @@ def create_package(
     setup_parsed = ParsedSetup.from_path(setup_directory_or_file)
 
     if setup_parsed.ext_modules:
-        run(
-            [sys.executable, "-m", "cibuildwheel", "--output-dir", dist], cwd=setup_parsed.folder
-        )
+        run([sys.executable, "-m", "cibuildwheel", "--output-dir", dist], cwd=setup_parsed.folder, check=True)
```
Comment (scbedd) [inline tools/azure-sdk-tools/ci_tools/build.py:183]: `check=True` was what I was missing before. Not certain how I messed that up.

File: tools/azure-sdk-tools/ci_tools/build.py [hunk 2]
```diff
@@ -180,17 +180,9 @@ def create_package(
...
 
     if enable_wheel:
-        run_logged(
-            [sys.executable, "setup.py", "bdist_wheel", "-d", dist], prefix="create_wheel", cwd=setup_parsed.folder
```
Comment (mccoyp) [inline tools/azure-sdk-tools/ci_tools/build.py:189]: Was removing the prefix here and for `create_sdist` intentional?

Comment (scbedd) [reply tools/azure-sdk-tools/ci_tools/build.py:189]: 100$ `prefix` doesn't apply to `run()`.

Comment (mccoyp) [reply tools/azure-sdk-tools/ci_tools/build.py:189]: LGTM then! Thanks for the clarification

**Review by mccoyp — APPROVED** [review]

**Review by jalauzon-msft — APPROVED** [review]
Approved for Storage. Thanks for getting to this!

**Pull request merged**